frontend/app/components/contextual-alert.tsx

```diff
@@ -0,0 +1,25 @@
+import type { ReactNode } from 'react';
+
+export interface ContextualAlertProps {
+  message_heading: string;
+  children: ReactNode;
+  type: 'warning' | 'success' | 'danger' | 'info';
+}
+
+export function ContextualAlert(props: ContextualAlertProps) {
+  const { message_heading, children, type } = props;
+
+  const alert_color = type === 'warning' ? 'border-l-amber-700' : type === 'danger' ? 'border-l-red-700' : type === 'info' ? 'border-l-cyan-700' : 'border-l-green-700';
```

Comment:
**Nitpick:** Some names here are using different casing than the rest of the application

```suggestion
  const { messageHeading, children, type } = props;

  const alertColour = type === 'warning' ? 'border-l-amber-700' : type === 'danger' ? 'border-l-red-700' : type === 'info' ? 'border-l-cyan-700' : 'border-l-green-700';
```